### DISCUSSION TITLE
File transfer in docker-compose setup

### DISCUSSION THREAD

**Discussion by LeventeLantos** [anacrolix/torrent#909] [Q&A]
Hi,
I am working on a pet project where I would like to only use a docker-compose stack for distributing data, this is my stack:
`version: '3.8'
services:
  distributor:
    build: .
    volumes:
      - ./distributor-data:/app/data
      - ./torrent-files:/app/torrent
    command: ["./distributor"]
    ports:
      - "8080:8080"
      - "42069:42069"
      - "6881:6881/udp"
    networks:
      vnet:

  receiver1:
    build: .
    volumes:
      - ./receiver1-data:/app/data
      - ./torrent-files:/app/torrent
    command: ["./receiver", "/app/data/myfile.torrent"]
    depends_on:
      - distributor
    networks:
      vnet:

  receiver2:
    build: .
    volumes:
      - ./receiver2-data:/app/data
      - ./torrent-files:/app/torrent
    command: ["./receiver", "/app/data/myfile.torrent"]
    depends_on:
      - distributor
    networks:
      vnet:

  receiver3:
    build: .
    volumes:
      - ./receiver3-data:/app/data
      - ./torrent-files:/app/torrent
    command: ["./receiver", "/app/data/myfile.torrent"]
    depends_on:
      - distributor
    networks:
      vnet:

networks:
  vnet:
    driver: bridge
`
I'm using magnet links, but I'm getting a lot of errors like this:
distributor-1  | [2024-02-29 15:58:00 +0000 DBG github.com/anacrolix/upnp upnp.go:189]
distributor-1  |   Listening for UPnP response for device type urn:schemas-upnp-org:device:InternetGatewayDevice:1 on eth0
distributor-1  | [2024-02-29 15:58:00 +0000 DBG dht [::]:42069 github.com/anacrolix/dht/v2 server.go:1255]
distributor-1  |   falling back on starting nodes
distributor-1  | 2024/02/29 15:58:00 Seeding myfile.txt...
distributor-1  | [2024-02-29 15:58:00 +0000 DBG github.com/anacrolix/torrent torrent.go:2103]
distributor-1  |   hashed piece 0 (passed=true)
distributor-1  | [2024-02-29 15:58:00 +0000 DBG dht [::]:42069 github.com/anacrolix/dht/v2 server.go:948]
distributor-1  |   Query(find_node) returned after 24.834µs (err=send 1: error writing 91 bytes to 212.129.33.59:6881: write udp6 [::]:42069->212.129.33.59:6881: sendto: network is unreachable, reply.Y=, reply.E=<nil>, writes=0)
distributor-1  | [2024-02-29 15:58:00 +0000 DBG dht 0.0.0.0:42069 github.com/anacrolix/dht/v2 server.go:948]
distributor-1  |   Query(find_node) returned after 44.792µs (err=send 1: error writing 91 bytes to [2001:19f0:5:6d01:5400:2ff:feec:644a]:6881: write udp4 0.0.0.0:42069->[2001:19f0:5:6d01:5400:2ff:feec:644a]:6881: address 2001:19f0:5:6d01:5400:2ff:feec:644a: non-IPv4 address, reply.Y=, reply.E=<nil>, writes=0)

Is it even possible to make this work inside a docker-compose setup?
Thanks in advance

**Comment by anacrolix** [comment]
Those are debugging messages, have you set Debug true in your ClientConfig?
